PowerValidatedSolutions.psm1

```diff
@@ -491,6 +491,81 @@ Function Set-vCenterPasswordExpiration {
+        This example configures all ESXi hosts within the cluster named sfo-m01-cl01 of the workload domain sfo-m01
+    #>
+	Param(
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$domain,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$cluster,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$ExpirationInDays,
+        [Parameter (Mandatory = $false)] [ValidateSet("true","false")] [String]$detail="true"
+	)
+	
+	Try{
+		if (Test-Connection -server $server) {
+			if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+				if (Get-VCFWorkloadDomain | Where-Object { $_.name -eq $domain }) {
+					if (($vcfVcenterDetails = Get-vCenterServerDetail -server $server -user $user -pass $pass -domain $domain)) {
+						if (Test-VsphereConnection -server $($vcfVcenterDetails.fqdn)) {
+							if (Test-VsphereAuthentication -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass) {
+								if (Get-Cluster | Where-Object { $_.Name -eq $cluster }) {
+									$esxiHosts = Get-Cluster $cluster | Get-VMHost
+									Foreach ($esxiHost in $esxiHosts) {
+										$passwordExpire = Get-VMHost -name $esxiHost | Where-Object { $_.ConnectionState -eq "Connected" -or $_.ConnectionState -eq "Maintenance" } | Get-AdvancedSetting | Where-Object { $_.Name -eq "Security.PasswordMaxDays" }
+										if ($passwordExpire) {
+											Set-AdvancedSetting -AdvancedSetting $passwordExpire -Value $ExpirationInDays -Confirm:$false | Out-Null
+											$checkSetting = Get-VMHost -name $esxiHost | Where-Object { $_.ConnectionState -eq "Connected" -or $_.ConnectionState -eq "Maintenance" } | Get-AdvancedSetting | Where-Object { $_.Name -eq "Security.PasswordMaxDays" }
+                                            if ($checkSetting -match $ExpirationInDays) {
+                                                if ($detail -eq "true") {
+                                                    Write-Output "Updating Advanced System Setting (Security.PasswordMaxDays) on ESXi Host ($esxiHost): SUCCESSFUL"
+                                                }
+                                            }
+                                            else {
+                                                Write-Error "Updating Advanced System Setting (Security.PasswordMaxDays) on ESXi Host ($esxiHost): POST_VALIDATION_FAILED"
+                                            }
+										}
+										else {
+											Write-Error "Error applying password policy"
```

Comment:
Does this need a `POST_VALIDATION_FAILED` msg?

Reply:
addressed in 430834ddd05e1b61bda2cc01157674496e856c62